test/cmdlineTests/viair_subobject_optimization/input.sol

```diff
@@ -0,0 +1,16 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity >0.0.0;
+
+contract C {
+  constructor(uint x) {
+    // The constant should be optimized away and occur neither in the bytecode of C
+    // nor embedded in the bytecode of D.
```

Comment:
Not sure if we have - or should come up with - a better way to test this...

It might even make sense to compile each and every semantics test contract directly and compare the bytecode with compiling it together with another contract that returns its ``type(...).creationCode``....